core/src/main/scala/com/avsystem/commons/concurrent/TaskExtensions.scala

```diff
@@ -0,0 +1,57 @@
+package com.avsystem.commons
+package concurrent
+
+import com.avsystem.commons.concurrent.TaskExtensions.{TaskCompanionOps, TaskOps}
+import com.avsystem.commons.misc.Timestamp
+import monix.eval.Task
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.FiniteDuration
+
+trait TaskExtensions {
+  implicit def taskOps[T](task: Task[T]): TaskOps[T] = new TaskOps(task)
+
+  implicit def taskCompanionOps(task: Task.type): TaskCompanionOps.type = TaskCompanionOps
+}
+
+object TaskExtensions extends TaskExtensions {
+  final class TaskOps[T](private val task: Task[T]) extends AnyVal {
+    /**
+      * Like regular `timeout` but [[TimeoutException]] is created lazily (for performance).
+      */
+    def lazyTimeout(after: FiniteDuration, msg: => String): Task[T] =
+      task.timeoutTo(after, Task.raiseError(new TimeoutException(msg)))
```

Comment:
This is still not lazy, you need `Task.defer( )` on top of the `Task.raiseError`

Reply:
Good point